keps/sig-api-machinery/4192-svm-in-tree/README.md

```diff
@@ -0,0 +1,484 @@
+
+###### What specific metrics should inform a rollback?
+The metric `storage_migrator_core_migrator_migrations_total` with `failed` status can indicate how many failures there are and can help decide if a rollback is required.
+
+The following metrics should not be available if we perform a rollback:
+- storage_migrator_core_migrator_migrated_objects_total
+- storage_migrator_core_migrator_remaining_objects_total
+- storage_migrator_core_migrator_migrations_total
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+This will be covered by integration tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+The following metrics are available when Storage Version Migration is enabled:
+- storage_migrator_core_migrator_migrated_objects_total
+- storage_migrator_core_migrator_remaining_objects_total
+- storage_migrator_core_migrator_migrations_total
+
+###### How can someone using this feature know that it is working for their instance?
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: **_MigrationCondition.Type (Running | Succeeded | Failed)_**
+  - Other field: **_MigrationCondition.LastUpdateTime_**
+- [x] Other (treat as last resort)
+  - Details:
+      -    The following metrics are available when Storage Version Migration is enabled:
+            - storage_migrator_core_migrator_migrated_objects_total
+            - storage_migrator_core_migrator_remaining_objects_total
+            - storage_migrator_core_migrator_migrations_total
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+NA
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [x] Other (treat as last resort)
+  - Details:
+      - The metric `storage_migrator_core_migrator_migrations_total` with the status `Failed` can be observed to determine the health of the migrations.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes. Creation of the Migration Request which creates `StorageVersionMigration` resource. 
```

Comment:
That's actually the least important stuff.

The most important stuff are the LIST requests that it's sending for resource types to be migrated and an Update request for each of the resources of that type to actually migrate it...

Reply:
Added it in the https://github.com/kubernetes/enhancements/pull/4276.